src/EPR.Calculator.Frontend/Controllers/CalculationRunDetailsController.cs

```diff
@@ -1,13 +1,152 @@
 
 namespace EPR.Calculator.Frontend.Controllers
 {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CalculationRunDetailsController"/> class.
+    /// </summary>
     public class CalculationRunDetailsController : Controller
     {
-        public IActionResult Index()
+        private readonly IConfiguration configuration;
+        private readonly IHttpClientFactory clientFactory;
+        private readonly ILogger<CalculationRunDetailsController> logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CalculationRunDetailsController"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration settings.</param>
+        /// <param name="clientFactory">The HTTP client factory.</param>
+        /// <param name="logger">The logger instance.</param>
+        public CalculationRunDetailsController(IConfiguration configuration, IHttpClientFactory clientFactory, ILogger<CalculationRunDetailsController> logger)
+        {
+            this.configuration = configuration;
+            this.clientFactory = clientFactory;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// Displays the calculation run details index view.
+        /// </summary>
+        /// <param name="runId">The ID of the calculation run.</param>
+        /// <param name="calcName">The calcName of the calculation run.</param>
+        /// <returns>The calculation run details index view.</returns>
+        public async Task<IActionResult> IndexAsync(int runId, string calcName)
+        {
+            try
+            {
+                var getCalculationDetailsResponse = await this.GetCalculationDetailsAsync(runId);
+
+                if (getCalculationDetailsResponse == null)
+                {
+                    this.logger.LogError(
+                            $"Request failed with status code {getCalculationDetailsResponse.StatusCode}");
+                    return this.RedirectToAction(ActionNames.StandardErrorIndex,
+                            CommonUtil.GetControllerName(typeof(StandardErrorController)));
+                }
+
+                var statusUpdateViewModel = new CalculatorRunStatusUpdateDto
+                {
+                    RunId = runId,
+                    ClassificationId = (int)RunClassification.DELETED,
+                    CalcName = calcName,
+                };
+
+                if (!getCalculationDetailsResponse.IsSuccessStatusCode)
+                {
+                    return this.RedirectToAction(ActionNames.StandardErrorIndex, CommonUtil.GetControllerName(typeof(StandardErrorController)));
+                }
+
+                return this.View(ViewNames.CalculationRunDetailsIndex, statusUpdateViewModel);
+            }
+            catch (Exception)
+            {
+                return this.RedirectToAction(ActionNames.StandardErrorIndex, CommonUtil.GetControllerName(typeof(StandardErrorController)));
```

Comment:
Can you please log as well as you already injecting logger but not using which will be good us for tracing. thanks